worlds/zork_grand_inquisitor/game_controller.py

```diff
@@ -0,0 +1,1364 @@
+        return locations_with_tag(ZorkGrandInquisitorTags.MISSABLE)
+
+    def log(self, message) -> None:
+        if self.logger:
+            self.logger.info(message)
+
+    def log_debug(self, message) -> None:
+        if self.logger:
+            self.logger.debug(message)
+
+    def open_process_handle(self) -> bool:
+        return self.game_state_manager.open_process_handle()
+
+    def close_process_handle(self) -> bool:
+        return self.game_state_manager.close_process_handle()
+
+    def is_process_running(self) -> bool:
+        return self.game_state_manager.is_process_running
+
+    def list_received_brog_items(self) -> None:
+        self.log("Received Brog Items:")
+
+        self._process_received_items()
+        received_brog_items: Set[ZorkGrandInquisitorItems] = self.received_items & self.brog_items
+
+        if not len(received_brog_items):
+            self.log("    Nothing")
+            return
+
+        for item in sorted(i.value for i in received_brog_items):
+            self.log(f"    {item}")
+
+    def list_received_griff_items(self) -> None:
+        self.log("Received Griff Items:")
+
+        self._process_received_items()
+        received_griff_items: Set[ZorkGrandInquisitorItems] = self.received_items & self.griff_items
+
+        if not len(received_griff_items):
+            self.log("    Nothing")
+            return
+
+        for item in sorted(i.value for i in received_griff_items):
+            self.log(f"    {item}")
+
+    def list_received_lucy_items(self) -> None:
+        self.log("Received Lucy Items:")
+
+        self._process_received_items()
+        received_lucy_items: Set[ZorkGrandInquisitorItems] = self.received_items & self.lucy_items
+
+        if not len(received_lucy_items):
+            self.log("    Nothing")
+            return
+
+        for item in sorted(i.value for i in received_lucy_items):
+            self.log(f"    {item}")
+
+    def list_received_hotspots(self) -> None:
+        self.log("Received Hotspots:")
+
+        self._process_received_items()
+
+        hotspot_items: Set[ZorkGrandInquisitorItems] = items_with_tag(ZorkGrandInquisitorTags.HOTSPOT)
+        received_hotspots: Set[ZorkGrandInquisitorItems] = self.received_items & hotspot_items
+
+        if not len(received_hotspots):
+            self.log("    Nothing")
+            return
+
+        for item in sorted(i.value for i in received_hotspots):
+            self.log(f"    {item}")
+
+    def update(self) -> None:
+        if self.game_state_manager.is_process_still_running():
+            try:
+                self.game_state_manager.refresh_game_location()
+
+                self._apply_permanent_game_state()
+                self._apply_conditional_game_state()
+
+                self._apply_permanent_game_flags()
+
+                self._check_for_completed_locations()
+
+                if self.option_grant_missable_location_checks is True:
```

Comment:
This one is more a force of habit. There is a difference in intent in checking for truthy / falsy values (especially since in Python, every type can be tested for truthiness) and specifically looking for True / False. Omitting `is True` is not just formatting shorthand.

I can change it, as it seems to bother Silvris too.